### ISSUE TITLE
RCS1146 issues

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#329]
Consider:

```csharp
public struct Money
{
    public decimal Amount { get; set; }

    public Money(decimal amount)
    {
        Amount = amount;
    }
}

public class Foo
{
    public int Bar { get; set; }
}

class Program
{
    static void Main(string[] args)
    {
        var m = (Money?)null;

        var amountPresent1 = m != null && m.Value.Amount > 0; // 1: no RCS1146; should suggest and rewrite to "m?.Amount > 0"

        var foo = (Foo)null;

        var amountPresent2 = amountPresent1 && foo != null && foo.Bar == 42; // 2: no RCS1146; should suggest and rewrite to "amountPresent1 && foo?.Bar == 42"
        var amountPresent3 = amountPresent1 && m != null && m.Value.Amount > 0; // 2: no RCS1146; should suggest and rewrite to "amountPresent1 && m?.Amount > 0"
    }
}
```

1. RCS1146 doesn't trigger for structs: Rosylnator does not suggest to rewrite `amountPresent1` using the C# 6.0 null propagation operator.
2. RCS1146 doesn't trigger when part of a complex expression: Rosylnator does not suggest to rewrite `amountPresent2` and `amountPresent3` using the C# 6.0 null propagation operator.
